### PR TITLE
CDAP-15030. Salesforce Streaming Source Integration Tests

### PR DESCRIPTION
**Pull request by aonischuk**
Integration tests do the following:
1. Start the pipeline with realtime streaming plugin as source and mockSink as sink.
2. Once the pipeline is started records are added to the remote salesforce instance specified in maven parameters
3. Than tests wait while these records are received by the pipeline.
4. The records than are asserted for correctness.

### CODE REVIEW

File: src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceStreamingSourceETLTest.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+  protected List<StructuredRecord> waitForRecords(ProgramManager programManager,
+                                                  int exceptedNumberOfRecords) throws Exception {
+    DataSetManager<Table> outputManager = getDataset(getOutputDatasetName());
+
+    try {
+      Awaitility.await()
+        .atMost(WAIT_FOR_RECORDS_TIMEOUT_SECONDS, TimeUnit.SECONDS)
+        .pollInterval(WAIT_FOR_RECORDS_POLLING_INTERVAL_MS, TimeUnit.MILLISECONDS)
+        .until(() -> MockSink.readOutput(outputManager).size() >= exceptedNumberOfRecords);
```
Comment (mlozbin-cybervisiontech) [inline src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceStreamingSourceETLTest.java]: maybe untilPassed?

File: src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceStreamingSourceETLTest.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      .build();
+
+    AppRequest<DataStreamsConfig> appRequest = new AppRequest<>(APP_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app(appName);
+    ApplicationManager applicationManager = deployApplication(appId, appRequest);
+
+    return applicationManager.getSparkManager(DataStreamsSparkLauncher.NAME);
+  }
+
+  private boolean hasReceiverThreadStarted() {
```
Comment (mlozbin-cybervisiontech) [inline src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceStreamingSourceETLTest.java]: We should not forget to think how to make same check in remoteInstance mode. Maybe wait for dummy message?

Comment (mlozbin-cybervisiontech) [inline src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceStreamingSourceETLTest.java]: at least provide an explanation of how it works and why it was done this way

**Review by mlozbin-cybervisiontech — CHANGES_REQUESTED** [review]

File: src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceStreamingSourceETLTest.java (hunk 1 above)
Comment (aonischuk) [reply src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceStreamingSourceETLTest.java]: do you mean we should create this method? Or it is some existent thing?

**Review by mlozbin-cybervisiontech — APPROVED** [review]

**Pull request merged**